test/rx/middleware_test.rb

```diff
@@ -51,6 +51,20 @@ def test_it_responds_to_deep_requests
     assert body[0] != "response"
   end
 
+  def test_deep_check_fails_if_default_authorization_fails
```

Comment:
this test is failing b/c of header names as well.

I feel like the right approach here is to just return a failed authorization instead of raising an exception though - passing the wrong header just means that the client isn't authenticated, right?

Reply:
Fixed. Regarding the approach, I was trying to handle the wrong header but you have a point here that it is still a failed authorization. in the new change, this part is removed